tests/unit/parser.js

```diff
@@ -4404,47 +4404,47 @@ exports["for of as legacy JS"] = function (test) {
-    .addError(8, 11, "Invalid for-of loop left-hand-side: initializer is forbidden.")
-    .addError(8, 11, "Invalid for-of loop left-hand-side: more than one ForBinding.")
-    .addError(9, 16, "'for of' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).")
-    .addError(9, 6, "Invalid for-of loop left-hand-side: initializer is forbidden.")
-    .addError(10, 15, "'for of' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).")
-    .addError(10, 11, "Invalid for-of loop left-hand-side: more than one ForBinding.")
-    .addError(11, 23, "'for of' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).")
-    .addError(11, 15, "Invalid for-of loop left-hand-side: initializer is forbidden.")
-    .addError(11, 15, "Invalid for-of loop left-hand-side: more than one ForBinding.")
-    .addError(12, 16, "'for of' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).")
-    .addError(12, 6, "Invalid for-of loop left-hand-side: initializer is forbidden.")
-    .addError(12, 6, "'let' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).")
-    .addError(13, 15, "'for of' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).")
-    .addError(13, 11, "Invalid for-of loop left-hand-side: more than one ForBinding.")
-    .addError(13, 6, "'let' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).")
-    .addError(14, 23, "'for of' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).")
-    .addError(14, 15, "Invalid for-of loop left-hand-side: initializer is forbidden.")
-    .addError(14, 6, "'let' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).")
-    .addError(14, 15, "Invalid for-of loop left-hand-side: more than one ForBinding.")
-    .addError(15, 18, "'for of' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).")
-    .addError(15, 6, "Invalid for-of loop left-hand-side: initializer is forbidden.")
-    .addError(15, 6, "'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).")
-    .addError(16, 17, "'for of' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).")
-    .addError(16, 13, "Invalid for-of loop left-hand-side: more than one ForBinding.")
-    .addError(16, 6, "'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).")
-    .addError(17, 25, "'for of' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).")
-    .addError(17, 17, "Invalid for-of loop left-hand-side: initializer is forbidden.")
-    .addError(17, 17, "Invalid for-of loop left-hand-side: more than one ForBinding.")
-    .addError(17, 6, "'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).")
-    .test(code, {undef: true, predef: ["print"]}); // es5
```

Comment:
These changes are invalid. The expected character number was added https://github.com/jshint/jshint/commit/e0d575f2803ff9d5891007f2c22c3953df4a21fe